src/Test.UnitTests.Sarif/UriConverterTests.cs

```diff
@@ -76,5 +83,72 @@ public void WriteJson_ConvertsUriObjectsToStrings()
 
             JsonConvert.SerializeObject(testObject, settings).Should().Be(s_testFileText);
         }
+
+        [Theory]
+        [InlineData("http://github.com/Microsoft/sarif-sdk")]
+        [InlineData("src/Program.cs")]
+        [InlineData("file:/src/Program.cs")]
+        [InlineData("file:/../../src/Program.cs")]
+        [InlineData("file://Code/src/Program.cs")]
+        [InlineData("file:///C:/Code/src/sarif-sdk/Program.cs")]
+        [InlineData("file:///new%2Dhost/share/folder/file.cs")]
+        [InlineData("file:///new%5Fhost/share/folder/file.cs")]
+        [InlineData("file:///new-host/share/folder%20with%20spaces/file.cs")]
+        [InlineData("file:///new-host/share/folder/%28surrounding-parens%29.cs")]
+        [InlineData("file:///%28new-host%29/share/folder/file.cs")]
+        [InlineData("file:///Local.wiki/WIKI/Engineering/Running%2DTests.md")]
+        [InlineData("http://www.example.com/dir/file.c")]
+        [InlineData("http://www.example.com/dir/file name.c")]
+        [InlineData("http://www.example.com/dir/file%20name.c")]
+        [InlineData(@"C:\dir\file.c")]
+        [InlineData(@"C:\dir\file name.c")]
+        [InlineData("/dir/file.c")]
+        [InlineData("/dir/file name.c")]
+        [InlineData("/dir/file%20name.c")]
+        [InlineData("file:///C:/dir/file.c")]
+        [InlineData("file:///C:/dir/file name.c")]
+        [InlineData("file:///C:/dir/file%20name.c")]
+        [InlineData(@"dir\file.c")]
+        [InlineData("dir/file.c")]
+        [InlineData(@"dir\file name.c")]
+        [InlineData("dir/file%20name.c")]
+        [InlineData("dir/file name.c")]
+        [InlineData(@"..\..\.\.\..\dir1\dir2\file.c")]
+        [InlineData("../../../dir1/dir2/file.c")]
+        [InlineData(@"..\..")]
+        [InlineData("../..")]
+        public void Uri_RoundTripping(string value)
+        {
+            // Framework Bug: Uris with certain escaped unreserved characters are doubled by Uri.TryCreate
+            // https://github.com/dotnet/runtime/issues/36288
+
+            SarifRoundTrip(new Uri(value, UriKind.RelativeOrAbsolute));
+            //DirectRoundTrip(new Uri(value, UriKind.RelativeOrAbsolute));
+        }
+
+        private Uri SarifRoundTrip(Uri value)
+        {
+            // Put in a class with a Uri using the Sarif 'UriConverter'
+            SingleUri sample = new SingleUri();
+            sample.Uri = value;
+
+            // Serialize and Deserialize
+            string json = JsonConvert.SerializeObject(sample);
+            SingleUri roundTripped = JsonConvert.DeserializeObject<SingleUri>(json);
+
+            roundTripped.Uri.Should().Be(value);
+            return roundTripped.Uri;
```

Comment:
>return roundTripped.Uri; [](start = 12, length = 24)

Return value not used, needed?

Reply:
It's not critical. I follow this pattern so that if the test wants to assert other details of the output, it can easily do so.